File: Validate Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isValidBST(self, root):
        tree = []
        self.inOrder(root,tree)
                print tree
        valid = True
        for i in range(0,len(tree)-1) :
            if tree[i+1] > tree[i]:
                valid = True
            else:
                valid = False
                break
                return valid
       def inOrder(self,root,res):
               if root is None : 
            return
        self.inOrder(root.left,res)
        res.append(root.val)
        self.inOrder(root.right,res)
                                                    """
        :type root: TreeNode
        :rtype: bool
        """
        

File: Validate Binary Search Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> tree;
    void inOrder(TreeNode* root){
        if (root){
            inOrder(root->left);
            tree.push_back(root->val);
            inOrder(root->right);
                    }
    }
    bool isValidBST(TreeNode* root) {
           /* if (root){
        if (root->left&& root->right){
            return  isValidBST(root->left)&&isValidBST(root->right)&&root->left->val < root->val < root->right
                ->val;
        }else if (root->left&& !root->right){
            return  isValidBST(root->left)&&root->left->val < root->val;
        }else if (!root->left&&root->right){
                        return isValidBST(root->right)&&root->right->val > root->val;
        }else{
            return 1;
        }
    }*/
        inOrder(root);
        for(int i = 1 ; i<tree.size();i++){
            if(tree[i] <=tree[i-1])
                return false;
        }
        return true;
            }
};